### PR TITLE
DSC_CMReportingServicePoint:  A resource to configure the SCCM Reporting Service Point

### PR DESCRIPTION
**Pull request by NEllis280**
#### Pull Request (PR) description
Adding a resource to configure the SCCM Reporting Service Point

#### This Pull Request (PR) fixes the following issues
None

#### Task list
- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [x] Resource documentation added/updated in README.md.
- [x] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/38)

### CODE REVIEW

File: tests/Unit/CMReportingServicePoint.tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+                Mock -CommandName Set-Location
+                Mock -CommandName Get-CMSiteSystemServer
+                Mock -CommandName New-CMSiteSystemServer
+                Mock -CommandName Add-CMReportingServicePoint
+                Mock -CommandName Set-CMReportingServicePoint
+                Mock -CommandName Remove-CMReportingServicePoint
+            }
+
+            Context 'When Set-TargetResource runs successfully' {
+
```
[the remark this answers is not in the document]
Comment (NEllis280) [reply tests/Unit/CMReportingServicePoint.tests.ps1]: Removed

File: tests/Unit/CMReportingServicePoint.tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+                        @{
+                            PropertyName = 'DatabaseName'
+                            Value2       = 'CM_LAB'
+                        }
+                        @{
+                            PropertyName = 'DatabaseServerName'
+                            Value2       = 'CA01.contoso.com'
+                        }
+                        @{
+                            PropertyName = 'UserName'
```
[the remark this answers is not in the document]
Comment (NEllis280) [reply tests/Unit/CMReportingServicePoint.tests.ps1:57]: Fixed throughout, but I left this particular one in order to match the casing on the actual prop. I'll leave this comment open so we can chat about it sometime today.

File: tests/Unit/CMReportingServicePoint.tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+                        }
+                    )
+                }
+
+                Mock -CommandName Import-ConfigMgrPowerShellModule
+                Mock -CommandName Set-Location
+            }
+
+            Context 'When retrieving reporting service point settings' {
+
```
[the remark this answers is not in the document]
Comment (NEllis280) [reply tests/Unit/CMReportingServicePoint.tests.ps1]: Removed

File: tests/Unit/CMReportingServicePoint.tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+{
+    InModuleScope $script:dscResourceName {
+        Describe 'ConfigMgrCBDsc - DSC_CMReportingServicePoint\Get-TargetResource' -Tag 'Get'{
+            BeforeAll{
+                $getInput = @{
+                    SiteCode       = 'Lab'
+                    SiteServerName = 'CA01.contoso.com'
+                }
+
+                $getRSPReturn = @{
```
[the remark this answers is not in the document]
Comment (NEllis280) [reply tests/Unit/CMReportingServicePoint.tests.ps1]: changed

File: source/DSCResources/DSC_CMReportingServicePoint/DSC_CMReportingServicePoint.psm1 [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+
+    if ($rspProps)
+    {
+        foreach ($rspProp in $rspProps)
+        {
+            switch ($rspProp.PropertyName)
+            {
+                'DatabaseName'         { $dbName = $rspProp.Value2 }
+                'DatabaseServerName'   { $dbServerName = $rspProp.Value2 }
+                'UserName'             { $account = $rspProp.Value2 }
```
[the remark this answers is not in the document]
Comment (NEllis280) [reply source/DSCResources/DSC_CMReportingServicePoint/DSC_CMReportingServicePoint.psm1:51]: Fixed throughout, but I left this particular one in order to match the casing on the actual prop. I'll leave this comment open so we can chat about it sometime today.

File: README.md [hunk 6]
```diff
@@ -417,3 +419,33 @@ Please check out common DSC Community [contributing guidelines](https://dsccommu
...
+- **[String] DatabaseServerName** _(Write)_: Specifies the name of the Configuration
+  Manager database server that you want to use as the data source for reports from
+  Microsoft SQL Server Reporting Services.
+  To specify a database instance, use the format Server Name\Instance Name.
+- **[String] FolderName** _(Write)_: Specifies the name of the report folder on
+  the report server. This parameter can only be used when installing the role.
+- **[String] ReportServerInstance** _(Write)_: Specifies the name of an instance
+  of Microsoft SQL Server Reporting Services. This parameter can only be used
+  when installing the role.
+- **[String] UserName** _(Write)_: Specifies a user name for an account that
```
[the remark this answers is not in the document]
Comment (NEllis280) [reply README.md]: Fixed

File: README.md [hunk 7]
```diff
@@ -417,3 +419,33 @@ Please check out common DSC Community [contributing guidelines](https://dsccommu
 - [CMSoftwareUpdatePoint_Absent](Source\Examples\Resources\CMSoftwareUpdatePoint\CMSoftwareUpdatePoint_Absent.ps1)
 - [CMSoftwareUpdatePoint_CMG](Source\Examples\Resources\CMSoftwareUpdatePoint\CMSoftwareUpdatePoint_CMG.ps1)
 - [CMSoftwareUpdatePoint_Present](Source\Examples\Resources\CMSoftwareUpdatePoint\CMSoftwareUpdatePoint_Present.ps1)
+
+### CMReportingServicePoint
+
+- **[String] SiteCode** _(Key)_: Specifies the Site Code for the Configuration
+  Manager site.
+- **[String] SiteServerName** _(Required)_: Specifies the Site Server to install
```
[the remark this answers is not in the document]
Comment (NEllis280) [reply README.md]: Fixed

File: source/DSCResources/DSC_CMReportingServicePoint/DSC_CMReportingServicePoint.schema.mof [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+class DSC_CMReportingServicePoint: OMI_BaseResource
+{
+    [Key, Description("Specifies the SiteCode for the Configuration Manager site.")] String SiteCode;
+    [Key, Description("Specifies the Site Server to install or configure the role on.")] String SiteServerName;
+    [Write, Description("Specifies the name of the Configuration Manager database that you want to use as the data source for reports from Microsoft SQL Server Reporting Services.")] String DatabaseName;
+    [Write, Description("Specifies the name of the Configuration Manager database server that you want to use as the data source for reports from Microsoft SQL Server Reporting Services.")] String DatabaseServerName;
+    [Write, Description("Specifies the name of the report folder on the report server.")] String FolderName;
+    [Write, Description("Specifies the name of an instance of Microsoft SQL Server Reporting Services.")] String ReportServerInstance;
+    [Write, Description("Specifies a user name for an account that Configuration Manager uses to connect with Microsoft SQL Server Reporting Services and that gives this user access to the site database.")] String UserName;
```
[the remark this answers is not in the document]
Comment (NEllis280) [reply source/DSCResources/DSC_CMReportingServicePoint/DSC_CMReportingServicePoint.schema.mof]: Changed here, in synopsis in the psm1 and in the readme.

File: source/DSCResources/DSC_CMReportingServicePoint/en-US/DSC_CMReportingServicePoint.strings.psd1 [hunk 9]
```diff
@@ -0,0 +1,12 @@
+ConvertFrom-StringData @'
+    RetrieveSettingValue = Getting information for the specified Reporting Service Point.
+    RSPNotInstalled      = Reporting Service Point is not installed on server: {0}.
+    RSPAbsent            = {0} Reporting Service Point expected absent returned Present.
+    TestState            = Test-TargetResource compliance check returned: {0}.
+    SiteServerRole       = {0} is not currently a site system server adding site system role.
+    AddRSPRole           = Adding Reporting Service Point role to {0}.
+    SettingValue         = Setting value: {0} to {1}.
+    RemoveRSPRole        = Removing Reporting Service Point role from {0}.
+    SpecifyUser          = The UserName parameter is required when installing the Reporting Service Point Role.
```
[the remark this answers is not in the document]
Comment (NEllis280) [reply source/DSCResources/DSC_CMReportingServicePoint/en-US/DSC_CMReportingServicePoint.strings.psd1]: Changed here and in tests -ExpectedMessage.

File: source/DSCResources/DSC_CMReportingServicePoint/en-US/DSC_CMReportingServicePoint.strings.psd1 [hunk 10]
```diff
@@ -0,0 +1,12 @@
+ConvertFrom-StringData @'
+    RetrieveSettingValue = Getting information for the specified Reporting Service Point.
+    RSPNotInstalled      = Reporting Service Point is not installed on server: {0}.
```
[the remark this answers is not in the document]
Comment (NEllis280) [reply source/DSCResources/DSC_CMReportingServicePoint/en-US/DSC_CMReportingServicePoint.strings.psd1:3]: Changed here and in the psm1

File: source/DSCResources/DSC_CMReportingServicePoint/DSC_CMReportingServicePoint.psm1 (hunk 5 above)
[the remark this answers is not in the document]
Comment (NEllis280) [reply source/DSCResources/DSC_CMReportingServicePoint/DSC_CMReportingServicePoint.psm1]: Changed and pushed.

File: tests/Unit/CMReportingServicePoint.tests.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (NEllis280) [reply tests/Unit/CMReportingServicePoint.tests.ps1]: Changed and pushed.

File: source/DSCResources/DSC_CMReportingServicePoint/en-US/DSC_CMReportingServicePoint.strings.psd1 (hunk 10 above)
Comment (rchristman89) [inline source/DSCResources/DSC_CMReportingServicePoint/en-US/DSC_CMReportingServicePoint.strings.psd1:3]: RspNotInstalled still looks wrong.

File: source/DSCResources/DSC_CMReportingServicePoint/DSC_CMReportingServicePoint.psm1 [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+
+    try
+    {
+        if ($Ensure -eq 'Present')
+        {
+            $evalList = @('DatabaseName','DatabaseServerName','Username','FolderName','ReportServerInstance')
+
+            foreach ($param in $PSBoundParameters.GetEnumerator())
+            {
+            if ($evalList -contains $param.key)
```
Comment (rchristman89) [inline source/DSCResources/DSC_CMReportingServicePoint/DSC_CMReportingServicePoint.psm1:167]: I believe tabs are off in this.

**Pull request merged**